### DISCUSSION TITLE
Laser 128

### DISCUSSION THREAD

**Discussion by tiktok4321** [rowokii/A2C_TangyPack#3] [Ideas]
I have a Laser 128.  The Laser has this same port, but utilizes many of the unused pins to fully support RGB natively.  Can you take advantage of this pinout for a more efficient port to VGA/DVI/HDMI?

**Comment by rowokii** [comment]
Well, I'm looking to see if I can support the Laser 128 with the Tangy Pack.    I have the pinout of the IIc and Laser 128 listed below.  The column marked TP is the digital signals the Tangy Pack is physically connected to. 

1.  I should be able to get monochrome mode working with the Laser 128 at a minimum.  

2.  I might be able to get color mode working with the Laser 128, but the biggest problem is the Laser 128 does not have the CREF signal which is the color reference signal that the IIc sends out.  If I can find a reference point in the video signal that does not change, I might be able to make an assumption about how the cref signal looks but this reference point has to be the same between power cycle, or the color will be messed up depending on power cycle state.  This part needs to be further explored.  Also, another limitation of the Laser 128 signaling is there is no TEXT or GR signal that tells the IIc to kill the color when displaying text.   So, it will always be displaying color unless you change to  monochrome mode manually.   

Looking at the RGBI signal of the Laser 128 is not possible with the current hardware since not all the signals are physically connected to the FPGA.  

As a complete side feature, I might be able to generate a straight RGB looking image if that's what you are looking for,  Actually, this part is just a straight up color look up table and is easier than what I'm doing to imitate the NTSC color fringe.    Hum... if I can get #2 working, maybe can I make this the default color mode for Laser 128.

PIN | TP | IIc         | Laser 128
--- | --- | ---| ---
1    |      | Text       | Intensity
2    |   * | 14M      | 14M
3    |   * | SYNC     | RED
4    |     | SEGB      | BLUE
5    |     | SOUND  | SOUND
6    |      | LDPS     | LDPS
7    | *   | WNDW  | DSPEN
8    |     | 12V        | 12V
9    |     | PRAS      | HSYNC
10  |  *  | GR         | CSYNC
11  | *   | SEROUT | SEROUT
12  |     | C. VID    | C. VID
13  |     | GND      | GND
14  |     | VIDD7   | VSYNC
15  | *  | CREF      | GREEN

**Comment by rowokii** [comment]
Hi, just FYI, Laser 128 support added with Firmware V1.10
https://github.com/rowokii/A2C_TangyPack/blob/main/Laser128.md
